server/src/ask.rs

```diff
@@ -95,8 +95,8 @@ pub(super) struct Question {
 
 pub(super) async fn ask(
     Path(eid): Path<Uuid>,
+    State(dynamo): State<Backend>,
     q: Json<Question>,
-    Extension(dynamo): Extension<Backend>,
```

Comment:
Why reorder these?

Reply:
axum 0.6 requires that only one extractor consumes the request body. Attempting to consume the body multiple times was a subtle source of bugs/confusion in 0.5.

It does this by requiring that only the last function argument implements [`FromRequest`](https://docs.rs/axum/latest/axum/extract/trait.FromRequest.html) (which can access the body) and the rest implement [`FromRequestParts`](https://docs.rs/axum/latest/axum/extract/trait.FromRequestParts.html) (which cannot). Looking at [`Handler`](https://docs.rs/axum/latest/axum/handler/trait.Handler.html#impl-Handler%3C(M%2C%20T1%2C%20T2%2C%20T3%2C%20T4%2C%20T5%2C%20T6%2C%20T7)%2C%20S%2C%20B%3E-for-F) might make it more clear.

That is documented [here](https://docs.rs/axum/latest/axum/extract/index.html#the-order-of-extractors).

So therefore `Json` needs to be last since it consumes the request body.

Reply:
Makes sense, thanks!